qiskit/transpiler/passes/optimization/optimize_1q_gates.py

```diff
@@ -174,13 +185,21 @@ def run(self, dag):
                                                  (2 * np.pi)) == 0:
                     right_name = "nop"
 
+            if right_name == "u1" and "u1" not in self.basis:
+                right_name = "u3"
+            if right_name == "u2" and "u2" not in self.basis:
+                right_name = "u3"
+
             new_op = Gate(name="", num_qubits=1, params=[])
-            if right_name == "u1":
+            if right_name == "u1" and "u1" in self.basis:
```

Comment:
Is the check above sufficient to know that `right_name == 'u1'` here already implies `'u1' in self.basis`?

Reply:
you are right. removed in dd3ae8c70